client/cache/read_through.go

```diff
@@ -0,0 +1,67 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"github.com/beego/beego/v2/core/berror"
+	"strings"
+	"time"
+)
+
+// readThroughCache is a decorator
+// add the read through function to the original Cache function
+type readThroughCache struct {
+	Cache
+	Expiration time.Duration
+	LoadFunc   func(ctx context.Context, key string) (any, error)
+}
+
+// NewReadThroughCache create readThroughCache
+func NewReadThroughCache(cache Cache, expiration time.Duration,
+	loadFunc func(ctx context.Context, key string) (any, error)) (Cache, error) {
+	if loadFunc == nil {
+		return nil, berror.Error(InvalidLoadFunc, "loadFunc cannot be nil")
+	}
+	return &readThroughCache{
+		Cache:      cache,
+		Expiration: expiration,
+		LoadFunc:   loadFunc,
+	}, nil
+}
+
+// Get cache from readThroughCache
```

Comment:
Get will try to call the LoadFunc to load data if the Cache returns value nil or non-nil error.